oauth.py

```diff
@@ -0,0 +1,141 @@
+"""
+Following is a python script for Oauth2 implementation of google.The same flow is applicable for any other oauth provider.
+
+Firstly, the user is redirected to authorization link which is google's sign in page where the user can sign in with his google account and then grant permission to your app to access their data.
+
+Then, the user is redirected back to redirected_url  with a authorization code appended in it.
+
+This auth code is then send with a post request to authorization server to get back an access token.
+
+The access token then can be send to resource server to get back the user information.
+
+"""
+
+
+from config import (CLIENT_ID,
+					CLIENT_SECRET,
+					AUTHORIZE_URL,
+					CALLBACK_URL,
+					ACCESS_TOKEN_URL,
+					API_RESOURCE_URL,
+					SCOPE_URL)
+from flask import Flask, request
+import requests
+import datetime
+import logging
+
+logging.basicConfig(filename='oauth.log',level=logging.ERROR)
+
+app = Flask(__name__)
+@app.route('/')
+
+def homepage():
+	"""
+	Displays the authorization link to users.
+
+	"""
+	text = '<a href="%s">Login with google</a>'
+	return text % get_authorization_url()
+
+
+def get_authorization_url():
+
+	"""
+	this function returns the authorization url
+
+	it makes a get request to the authorization 
+	url of google and  with the parameters
```

Comment:
Instead of so many verbose comments, add request dict/sample data wherever necessary